File: backend/WarehouseOps.Domain/Customer.cs
namespace WarehouseOps.Domain;

public class Customer : BaseEntity
{
    public string Name { get; set; } = string.Empty;

    public string Email { get; set; } = string.Empty;

    public string PhoneNumber { get; set; } = string.Empty;

    public string Address { get; set; } = string.Empty;

    public List<Order> Orders { get; set; } = new();
}

File: frontend/warehouseops-client/src/types/customer.ts
export type Customer = {
  id: string;
  name: string;
  email: string;
  phoneNumber: string;
  address: string;
  createdAt: string;
  updatedAt: string | null;
};

export type CustomerFilters = {
  search?: string;
};

export type CreateCustomerRequest = {
  name: string;
  email: string;
  phoneNumber: string;
  address: string;
};

export type UpdateCustomerRequest = {
  name: string;
  email: string;
  phoneNumber: string;
  address: string;
};
